src/layout/dom.ts

```diff
@@ -199,14 +201,20 @@ function selector(id: number, data: NodeInfo, parent: string): string {
         case "*D":
             return "";
         default:
+            // Do not compute hash for nodes that are disconnected from DOM (i.e. no parent)
+            // The only exception is HTML element, which is the root element of DOM tree
+            if (parent === null && data.tag !== "HTML") { return ""; }
+
+            let parentSelector = parent ? parent.selector : "";
             let value = getValue(id);
             let ex = value ? value.selector : null;
             let attributes = "attributes" in data ? data.attributes : {};
-            let current = "id" in attributes ? `${data.tag}#${attributes.id}` : `${parent}>${data.tag}`;
-            if ("class" in attributes) { current = `${current}.${attributes.class.trim().split(" ").join(".")}`; }
-            if (ID_ATTRIBUTE in attributes) { current = `*${attributes[ID_ATTRIBUTE]}`; }
-            if (current !== ex && selectorMap.indexOf(id) === -1) { selectorMap.push(id); }
-            return current;
+            let tag = data.tag.indexOf(SVG_PREFIX) === 0 ? data.tag.substr(SVG_PREFIX.length) : data.tag;
+            let s = "id" in attributes && attributes["id"].length > 0 ? `${tag}#${attributes.id}` : `${parentSelector}>${tag}`;
```

Comment:
No change but worth discussing this logic given how important it is for analytics.

Reply:
This is all based on Swaathee's feedback as she is prototyping Heatmaps.

Reply:
And, hoping to discuss this more as part of the design review later.